application/controllers/Notes.php

```diff
@@ -1,119 +1,324 @@
+    // Ensure only authorized users can access Notes controller
+    function __construct() {
+        parent::__construct();
+        $this->load->model('user_model');
+        if (!$this->user_model->authorize(2)) {
+            $this->session->set_flashdata('error', __("You're not allowed to do that!"));
+            redirect('dashboard');
+            exit;
+        }
+    }
 
-	function __construct()
-	{
-		parent::__construct();
+	// Main notes page: lists notes and categories
+    public function index() {
+        $this->load->model('note');
+        $data = [];
+        // Get all notes for logged-in user
+        $data['notes'] = $this->note->list_all();
+        // Get possible categories
+        $data['categories'] = Note::$possible_categories;
+        // Get note counts per category
+        $category_counts = [];
+        foreach ($data['categories'] as $cat) {
+            $category_counts[$cat] = $this->note->count_by_category($cat);
+        }
+        $data['category_counts'] = $category_counts;
+        // Get total notes count
+        $data['all_notes_count'] = $this->note->count_by_category();
+        $data['page_title'] = __("Notes");
+        // Render views
+        $this->load->view('interface_assets/header', $data);
+        $this->load->view('notes/main', $data);
+        $this->load->view('interface_assets/footer');
+    }
 
-		$this->load->model('user_model');
-		if(!$this->user_model->authorize(2)) { $this->session->set_flashdata('error', __("You're not allowed to do that!")); redirect('dashboard'); }
-	}
+	// Add a new note
+    function add() {
+        $this->load->model('note');
+        $this->load->library('form_validation');
+        $suggested_title = null;
+        // Validate form fields
+        $this->form_validation->set_rules('title', 'Note Title', 'required|callback_contacts_title_unique');
+        $this->form_validation->set_rules('content', 'Content', 'required');
+        if ($this->form_validation->run() == FALSE) {
+            // If validation failed and Contacts, get suggested title
+            $cat = $this->input->post('category', TRUE);
+            if ($cat === 'Contacts') {
+                $this->load->helper('callsign');
+                $parsed = parse_callsign($this->input->post('title', TRUE));
+                $suggested_title = strtoupper($parsed['core']);
+            }
+            $data['suggested_title'] = $suggested_title;
+            $data['page_title'] = __("Add Notes");
+            $this->load->view('interface_assets/header', $data);
+            $this->load->view('notes/add', $data);
+            $this->load->view('interface_assets/footer');
+        } else {
+            // Ensure title is uppercase for Contacts before saving
+            $cat = $this->input->post('category', TRUE);
+            $title = $this->input->post('title', TRUE);
+            if ($cat === 'Contacts') {
+                $_POST['title'] = strtoupper($title);
```

Comment:
Good evening
just scrolled through and noticed this little security issue. Will go through this PR tomorrow in more detail, even it's a draft yet. 

The issue here is that you request the POST data again - and you call it with the PHP native method which does not include any XSS cleaning. Therefore this could be a vulnerability.

Simply call the variable instead, you already defined it in Ln 64



More tomorrow 👍😄
```suggestion
                $title = strtoupper($title);
```